pkg/capisdk/capi_gateway_manager.go

```diff
@@ -0,0 +1,197 @@
+package capisdk
+
+import (
+	"context"
+	"fmt"
+	"github.com/vmware/cloud-provider-for-cloud-director/pkg/vcdsdk"
+	"k8s.io/klog"
+)
+
+type CapvcdGatewayManager struct {
+	gatewayManager *vcdsdk.GatewayManager
+}
+
+func NewCapvcdGatewayManager(ctx context.Context, client *vcdsdk.Client, networkName string, ipamSubnet string) (*CapvcdGatewayManager, error) {
+	gatewayManager, err := vcdsdk.NewGatewayManager(ctx, client, networkName, ipamSubnet)
+	if err != nil {
+		return nil, fmt.Errorf("error creating gateway manager: [%v]", err)
+	}
+	return &CapvcdGatewayManager{
+		gatewayManager: gatewayManager,
+	}, nil
+}
+
+func (capvcdGatewayManager *CapvcdGatewayManager) CreateL4LoadBalancer(ctx context.Context, virtualServiceNamePrefix string,
```

Comment:
Can't we merge this functionality into the HTTP code itself? There is already L4 getting handled along with HTTP/HTTPS in CPI code, so potentially this function could be removed.